#!/usr/bin/env python
PACKAGE = "iiwa_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gain0", double_t,    0,     "damping eigenvalue 0", 200,  0, 10000)
gen.add("gain1", double_t,    0,     "damping eigenvalue 1",  200,  0, 10000)
gen.add("rotationalStiffness",   double_t,    0,     "set rotation error stiffness", 30,  0, 500)
gen.add("rotationalDamping",     double_t,    0.05,  "set rotation error stiffness", 3,  0, 500)
gen.add("useNullSpace", 	   bool_t,      0,     "Use null space to satisfy additional constraints",True)
gen.add("jointLimitsGain",      double_t,    0,  "Gain for joints limit constraint", 1,  0, 5)
gen.add("desiredJointsGain",      double_t,    0,  "Gain for desired joints constraint", 0,  0, 5)
gen.add("jointVelocitiesGain",      double_t,    0,  "Gain for joint velocities constraint", 0.01,  0, 1)

exit(gen.generate(PACKAGE, "iiwa_control", "DSImpedance_param"))
